controllers/users.ts

```diff
@@ -35,4 +36,32 @@ const getSelfData = (
   }
 };
 
-export { getSelfData };
+const patchSelfData = async (
+  req: Request,
+  res: Response
+): Promise<Response<any, Record<string, any>> | Express.BoomError<null>> => {
+  try {
+    const userId = req.userData?.id;
+    const { firstname, lastname, bio, username } = req.body;
+    await prisma.users.update({
+      where: {
+        id: userId,
+      },
+      // Allowed fields to edit from patch call
+      data: {
```

Comment:
@prakashchoudhary07 , will this function also update the firstname and other fields even though those are not passed? In that case, those will be updated to `null`. Can you verify this?

Reply:
Those won't be updated, have verified it